cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/GlobalParam.scala

```diff
@@ -0,0 +1,304 @@
+package com.microsoft.azure.synapse.ml.services
+
+import com.microsoft.azure.synapse.ml.param.ServiceParam
+import org.apache.spark.ml.param.Param
+import org.apache.spark.sql.Row
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+trait GlobalKey[T] {
+  val name: String
+  val isServiceParam: Boolean
+}
+
+case object OpenAIDeploymentNameKey extends GlobalKey[String]
+  {val name: String = "OpenAIDeploymentName"; val isServiceParam = true}
```

Comment:
Can this be defined in tests classes instead?

Reply:
I can't move the keys anywhere else because of the mapping from strings to keys that we have in GlobalParams. It's not ideal :(

Reply:
With the new changes, yes! Have moved all of the Keys to their appropriate locations